src/passes/rejectUnsupportedFeatures.ts

```diff
@@ -174,6 +174,31 @@ export class RejectUnsupportedFeatures extends ASTMapper {
     } else if (node.kind === FunctionKind.Receive) {
       throw new WillNotSupportError(`Receive functions are not supported`, node);
     }
+
+    //Checking if the function defination is constructor and
+    // if it is, checking if 'this' keyword is used to call
+    // the external/public functions of the contract
+    const nodesWithThisIdentifier: ASTNode[] | undefined =
+      node.kind === FunctionKind.Constructor
+        ? node.vBody
+            ?.getChildren()
+            .filter((value, _) => value instanceof Identifier && value.raw.name === 'this')
+        : undefined;
```

Comment:
This is a nitpick. If the sole point of this variable is to be defined when the function definition is a constructor, I think code readability would improve by swapping the ternary expression for an  if:
```Typescript
if (node.kind === FunctionKind.Constructor) {
 // Declare variables and write any logic related with constructors here
}
```